src/app/public/modules/text-editor/services/text-editor-adapter.service.ts

```diff
@@ -0,0 +1,532 @@
+  }
+
+  private static getColor(documentEl: Document, selector: string): string {
+    const commandValue = documentEl.queryCommandValue(selector);
+
+    // Edge is weird and returns numbers
+    if (typeof commandValue === 'number') {
+      /* istanbul ignore next */
+      // tslint:disable-next-line: no-bitwise
+      return 'rgb(' + (commandValue & 0xFF) + ', ' +
+                    // tslint:disable-next-line: no-bitwise
+                    ((commandValue & 0xFF00) >> 8) + ', ' +
+                    // tslint:disable-next-line: no-bitwise
+                    ((commandValue & 0xFF0000) >> 16 ) + ')';
+    }
+
+    // Firefox uses 'Transparent' instead of a color value
+    /* istanbul ignore next */
+    if (commandValue.toString().toLowerCase() === 'transparent') {
+      return STYLE_STATE_DEFAULTS.backColor;
+    }
+    return commandValue;
+  }
+
+  private hasLink(editorId: string): boolean {
+    const anchorEl = this.getSelectedAnchorTag(editorId);
+    const childAnchorEls = this.getChildSelectedAnchorTags(editorId);
+    return childAnchorEls.length > 0 || (!!anchorEl && !!anchorEl.href);
+  }
+
+  private static getParent(element: Element, tag: string): Element {
+    let currentNode = element;
+    while (currentNode && currentNode.tagName.toUpperCase() !== 'BODY') {
+      if (currentNode.tagName.toUpperCase() === tag.toUpperCase()) {
+        return currentNode;
+      }
+      currentNode = currentNode.parentElement;
+    }
+    return undefined;
+  }
+
+  private editorSelected(documentEl: Document): boolean {
+    const editor: HTMLElement = documentEl.body;
+    const selectedNode = this.selectionService.getCurrentSelection(documentEl).anchorNode;
+    /* istanbul ignore next */
+    return selectedNode &&
+      (
+        editor.contains(selectedNode) ||
+        (selectedNode.parentNode && editor.contains(selectedNode.parentNode))
+      );
+  }
+
+  /* istanbul ignore next */
+  private insertHtmlInIE11(id: string, html: string): void {
+    const documentEl = this.getDocumentEl(id);
+    const windowEl = this.getContentWindowEl(id);
+    const sel = windowEl.getSelection();
+    if (sel.getRangeAt && sel.rangeCount) {
+        let range = sel.getRangeAt(0);
+        range.deleteContents();
+
+        const el = documentEl.createElement('div');
+        el.innerHTML = html;
+        let frag = documentEl.createDocumentFragment(), node, lastNode;
+        while (el.firstChild) {
+          node = el.firstChild;
+          lastNode = frag.appendChild(node);
+        }
+        range.insertNode(frag);
+
+        // Preserve the selection
+        if (lastNode) {
+            range = range.cloneRange();
+            range.setStartAfter(lastNode);
+            range.collapse(true);
+            sel.removeAllRanges();
+            sel.addRange(range);
+        }
+    }
+  }
+
+  private static cleanUpBlankStyleTags(doc: Document): void {
```

Comment:
Same private static question

Reply:
Talked offline - reverted to non-static methods.